Lameck1/r-detective
# Pull request for Feature/linter logic: r-detective development

### What does this do? 

Building Ruby Capstone Project, code linter written in the Ruby language

R-detective is a beginner code analysis tool that provides feedback about code errors. 

### Description of Tasks completed
These eight methods have been implemented:
      - detect_bad_comment_syntax
      - detect_bad_file_name
      - detect_proc_new_usage
      - detect_trailing_empty_lines
      - detect_source_file_too_long
      - detect_leading_empty_lines
      - detect_double_spaces
      - detect_trailing_spaces

The following recommendations have been implemented after the first review:

#### Implement TSEs recommendations
- Removed puts from class file and main.rb to be in line with OOP best
  practices
- Marked these two methods as private because they are not used outside
  the GetOffence class file: #report_offence and #line_position

### How should this be manually tested? 
* Clone this repo https://github.com/Lameck1/r-detective
* Navigate to project directory. ```cd r-detective```
* Run ```bundle install``` to install dependencies
* Lastly run ```bin/main.rb test_files/code_smells.rb``` to check for errors the file ```code_smells.rb``
